File: source/com/barrybecker4/simulation/liquid1/Logger.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.liquid1;

import com.barrybecker4.common.app.ILog;
import com.barrybecker4.ui.dialogs.OutputWindow;
import com.barrybecker4.ui.util.Log;

/**
 * Singleton instance of logger.
 *
 * @author Barry Becker
 */
public class Logger {

    /** for debugging */
    public static final int LOG_LEVEL = 0;

    /** The singleton instance */
    private static ILog logger;

    /**
     * Constructor
     */
    private Logger() {}

    public static ILog getInstance()  {
        if (logger == null)   {

            logger = new Log( new OutputWindow( "Log", null ) );
            logger.setDestination(Log.LOG_TO_WINDOW);
        }
        return logger;
    }

    public static void log(int level, String msg) {
       getInstance().println(level, LOG_LEVEL, msg);
    }
}

File: scala-source/com/barrybecker4/simulation/liquid/Logger.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.liquid

import com.barrybecker4.common.app.ILog
import com.barrybecker4.ui.dialogs.OutputWindow
import com.barrybecker4.ui.util.Log


/**
  * Singleton instance of logger.
  * @author Barry Becker
  */
object Logger {

  /** for debugging */
  val LOG_LEVEL = 0

  /** The singleton instance */
  private var logger: Log = null

  def getInstance: ILog = {
    if (logger == null) {
      logger = new Log(new OutputWindow("Log", null))
      logger.setDestination(Log.LOG_TO_WINDOW)
    }
    logger
  }

  def log(level: Int, msg: String): Unit = {
    getInstance.println(level, LOG_LEVEL, msg)
  }
}

class Logger private()
